lib/BasicEvaluatedExpression.js

```diff
@@ -2,156 +2,190 @@
+		else if(this.isNumber()) return !!this.number;
+		else if(this.isRegExp()) return true;
+		else if(this.isArray()) return true;
+		else if(this.isConstArray()) return true;
+		else if(this.isWrapped()) return this.prefix && this.prefix.asBool() || this.postfix && this.postfix.asBool() ? true : undefined;
+		else if(this.isTemplateString()) {
+			if(this.quasis.length === 1) return this.quasis[0].asBool();
+			for(var i = 0; i < this.quasis.length; i++) {
+				if(this.quasis[i].asBool()) return true;
+			}
+			// can't tell if string will be empty without executing
 		}
-		// can't tell if string will be empty without executing
-	}
-	return undefined;
-};
-BasicEvaluatedExpression.prototype.set = function(value) {
-	if(typeof value === "string") return this.setString(value);
-	if(typeof value === "number") return this.setNumber(value);
-	if(typeof value === "boolean") return this.setBoolean(value);
-	if(value === null) return this.setNull();
-	if(value instanceof RegExp) return this.setRegExp(value);
-	if(Array.isArray(value)) return this.setArray(value);
-	return this;
-};
-BasicEvaluatedExpression.prototype.setString = function(str) {
-	if(str === null)
-		delete this.string;
-	else
-		this.string = str;
-	return this;
-};
-BasicEvaluatedExpression.prototype.setNull = function() {
-	this.null = true;
-	return this;
-};
-BasicEvaluatedExpression.prototype.setNumber = function(num) {
-	if(num === null)
-		delete this.number;
-	else
-		this.number = num;
-	return this;
-};
-BasicEvaluatedExpression.prototype.setBoolean = function(bool) {
-	if(bool === null)
-		delete this.bool;
-	else
-		this.bool = bool;
-	return this;
-};
-BasicEvaluatedExpression.prototype.setRegExp = function(regExp) {
-	if(regExp === null)
-		delete this.regExp;
-	else
-		this.regExp = regExp;
-	return this;
-};
-BasicEvaluatedExpression.prototype.setIdentifier = function(identifier) {
-	if(identifier === null)
-		delete this.identifier;
-	else
-		this.identifier = identifier;
-	return this;
-};
-BasicEvaluatedExpression.prototype.setWrapped = function(prefix, postfix) {
-	this.prefix = prefix;
-	this.postfix = postfix;
-	return this;
-};
-BasicEvaluatedExpression.prototype.unsetWrapped = function() {
-	delete this.prefix;
-	delete this.postfix;
-	return this;
-};
-BasicEvaluatedExpression.prototype.setOptions = function(options) {
-	if(options === null)
-		delete this.options;
-	else
-		this.options = options;
-	return this;
-};
-BasicEvaluatedExpression.prototype.setItems = function(items) {
-	if(items === null)
-		delete this.items;
-	else
-		this.items = items;
-	return this;
-};
-BasicEvaluatedExpression.prototype.setArray = function(array) {
-	if(array === null)
-		delete this.array;
-	else
-		this.array = array;
-	return this;
-};
-BasicEvaluatedExpression.prototype.setTemplateString = function(quasis) {
-	if(quasis === null)
-		delete this.quasis;
-	else
-		this.quasis = quasis;
-	return this;
+		return undefined;
+	}
+
+	set(value) {
```

Comment:
Could you check if this method is used?

I think it's unused and we can remove it...